src/Profiles/Shifting/DataPages/Base/ANTPLUS_ShiftingBaseMainDataPageMsg.cpp

```diff
@@ -0,0 +1,42 @@
+#include <Profiles/Shifting/DataPages/Base/ANTPLUS_ShiftingBaseMainDataPageMsg.h>
+#include <Profiles/Shifting/ANTPLUS_ShiftingPrivateDefines.h>
+#include <ANTPLUS_PrivateDefines.h>
+
+ShiftingBaseMainDataPageMsg::ShiftingBaseMainDataPageMsg(uint8_t dataPageNumber) : BaseDataPageMsg<BroadcastDataMsg>()
+{
+    memset(_buffer, 0, MESSAGE_SIZE);
+    _buffer[ANTPLUS_SHIFTING_DATAPAGEBASE_DATAPAGE_BYTE] = dataPageNumber;
+    _buffer[2] = 0xFF;
+
+    setTotalNumbersGearFront(1);
```

Comment:
Are these default values in the profile?

Also we should use `setDataBuffer(_buffer)` here instead of the call in the profile, this keeps the logic hidden inside the class.

Reply:
Defaults removed, setBuffer() used once in constructor